frontend/src/pages/modelRegistrySettings/utils.ts

```diff
@@ -0,0 +1,47 @@
+import { RecursivePartial } from '~/typeHelpers';
+import { ModelRegistryKind } from '~/k8sTypes';
+import { CA_BUNDLE_CRT, ODH_CA_BUNDLE_CRT, ODH_TRUSTED_BUNDLE } from './const';
+import { SecureDBInfo, SecureDBRType } from './CreateMRSecureDBSection';
+
+export const findSecureDBType = (name: string, key: string): SecureDBRType => {
+  if (name === ODH_TRUSTED_BUNDLE && key === CA_BUNDLE_CRT) {
+    return SecureDBRType.CLUSTER_WIDE;
+  }
+  if (name === ODH_TRUSTED_BUNDLE && key === ODH_CA_BUNDLE_CRT) {
+    return SecureDBRType.OPENSHIFT;
+  }
+  return SecureDBRType.EXISTING;
+};
+
+export const findConfigMap = (secureDBInfo: SecureDBInfo): { name: string; key: string } => {
+  if (secureDBInfo.type === SecureDBRType.CLUSTER_WIDE) {
+    return { name: ODH_TRUSTED_BUNDLE, key: CA_BUNDLE_CRT };
+  }
+  if (secureDBInfo.type === SecureDBRType.OPENSHIFT) {
+    return { name: ODH_TRUSTED_BUNDLE, key: ODH_CA_BUNDLE_CRT };
+  }
+  return { name: secureDBInfo.resourceName, key: secureDBInfo.key };
+};
+
+export const constructRequestBody = (
+  data: RecursivePartial<ModelRegistryKind>,
+  secureDBInfo: SecureDBInfo,
+  addSecureDB: boolean,
+): RecursivePartial<ModelRegistryKind> => {
+  const mr = data;
+  if (addSecureDB && secureDBInfo.resourceType === 'Secret' && mr.spec?.mysql) {
+    mr.spec.mysql.sslRootCertificateSecret = {
+      name: secureDBInfo.resourceName,
+      key: secureDBInfo.key,
+    };
+    mr.spec.mysql.sslRootCertificateConfigMap = null;
+  } else if (addSecureDB && mr.spec?.mysql) {
+    mr.spec.mysql.sslRootCertificateConfigMap = findConfigMap(secureDBInfo);
+    mr.spec.mysql.sslRootCertificateSecret = null;
```

Comment:
I had a little trouble realizing which of these cases we'll hit if the user is uploading a new cert and whether it's okay, and I think we'll hit this one and `findConfigMap` will correctly use the `resourceName` that was set by the frontend for the configmap we haven't yet created. Thinking through it again I think it may have been better to let the cluster generate the name using `metadata.generateName` instead of passing around the name for something that doesn't exist yet, but I realize we needed this to implement @yih-wang 's design and I think what you have works well enough.

Just a thought, I'm not sure if there's any way to make the code more clear - it's probably fine as-is.